src/goto-instrument/code_contracts.cpp

```diff
@@ -961,3 +950,723 @@ bool code_contractst::enforce_contracts(
+    symbolt standin_symbol = contract.new_tmp_symbol(
+      operation_address.type(),
+      function_symbol.location,
+      function_id,
+      function_symbol.mode);
+
+    symbol_exprt current_standin = standin_symbol.symbol_expr();
+    local_standin_variables.push_back(current_standin);
+
+    // Add to standin variable initialization block
+    init_block.add(
+      goto_programt::make_decl(current_standin, function_symbol.location));
+    init_block.add(goto_programt::make_assignment(
+      code_assignt(current_standin, operation_address),
+      function_symbol.location));
+
+    if(current_operation.type().id() == ID_struct_tag)
+    {
+      const symbolt &current_struct_symbol =
+        contract.get_namespace().lookup(to_tag_type(current_operation.type()));
+
+      const struct_typet &curr_struct_t =
+        to_struct_type(current_struct_symbol.type);
+      for(struct_union_typet::componentt component : curr_struct_t.components())
+      {
+        exprt current_member = member_exprt(current_operation, component);
+        component_members.push_back(current_member);
+      }
+    }
+    component_members.erase(component_members.begin());
+  }
+}
+
+std::vector<symbol_exprt>
+assigns_clause_struct_targett::temporary_declarations() const
+{
+  return local_standin_variables;
+}
+
+exprt assigns_clause_struct_targett::alias_expression(const exprt &ptr)
+{
+  exprt running = false_exprt();
+  bool first_it = true;
+  for(symbol_exprt symbol : local_standin_variables)
+  {
+    typet ptr_concrete_type = dereference_exprt(ptr).type();
+    auto left_size = size_of_expr(ptr_concrete_type, contract.get_namespace());
+    typet standin_concrete_type = dereference_exprt(symbol).type();
+    auto right_size =
+      size_of_expr(standin_concrete_type, contract.get_namespace());
+    if(!left_size.has_value())
+    {
+      log.error().source_location = ptr.source_location();
+      log.error() << "unable to determine size of type (lhs): "
+                  << type2c(ptr_concrete_type, contract.get_namespace())
+                  << " and " << ptr_concrete_type.id() << messaget::eom;
+      throw 0;
+    }
+    if(!right_size.has_value())
+    {
+      log.error().source_location = ptr.source_location();
+      log.error() << "unable to determine size of type (rhs): "
+                  << type2c(standin_concrete_type, contract.get_namespace())
+                  << messaget::eom;
+      throw 0;
+    }
+    if(
+      left_size.value().get(ID_value) ==
+      right_size.value().get(ID_value).c_str())
+    {
+      exprt same_obj = same_object(ptr, symbol);
+      exprt same_offset =
+        equal_exprt(pointer_offset(ptr), pointer_offset(symbol));
+
+      const exprt &compatible = and_exprt(same_obj, same_offset);
+      if(first_it)
+      {
+        running = compatible;
+        first_it = false;
+      }
+      else
+      {
+        running = or_exprt(running, compatible);
+      }
```

Comment:
... (continues from above): replace this by `disjuncts.push_back(and_exprt{same_obj, same_offset});` ... (continues below)